sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/pubsub/PubsubSchemaCapableIOProvider.java

```diff
@@ -116,11 +116,21 @@ public Schema configurationSchema() {
   public PubsubSchemaIO from(String location, Row configuration, Schema dataSchema) {
     validateConfigurationSchema(configuration);
     validateDlq(configuration.getValue("deadLetterQueue"));
-    validateEventTimestamp(dataSchema);
+    validateDataSchema(dataSchema);
     return new PubsubSchemaIO(location, configuration, dataSchema);
   }
 
-  private void validateEventTimestamp(Schema schema) {
+  @Override
+  public Boolean requiresDataSchema() {
+    return true;
+  }
+
+  private void validateDataSchema(Schema schema) {
+    if (requiresDataSchema() && schema == null) {
```

Comment:
Consider removing `requiresDataSchema()` because it is always true here?

I know this is the general logic but in this specific provider this call is not necessary.